source/response.ts

```diff
@@ -7,6 +8,7 @@ export function handleResponseCode(context: WebDAVClientContext, response: Respo
     if (status >= 400) {
         const err: WebDAVClientError = new Error(`Invalid response: ${status} ${response.statusText}`) as WebDAVClientError;
         err.status = status;
+        err.response = response as AxiosResponse;
```

Comment:
I'd prefer not to expose Axios-specific properties outside of this library, as I can't guarantee that I'll always use Axios as the request client. Would you consider changing this to `err.response = response` (only) and using the `Response` type in the `WebDAVClientError` interface?